extension/src/popup/views/AddAccount/AddAccount/index.tsx

```diff
@@ -46,6 +50,12 @@ export const AddAccount = () => {
     }
   };
 
+  React.useEffect(() => {
+    return function cleanup() {
+      dispatch(clearApiError());
```

Comment:
Any reason to not just `return () => {`?

It would prevent you from having to change the eslint rule

Reply:
hmm, it didn't seem to. I believe it's complaining about the use of the curly brackets in the return instead of an implicit return but if I do that then I get a pretty cryptic type error that I haven't understood yet.

Reply:
I did find one way to avoid removing the rule which is by doing this -

```
useEffect(() => {
    return () => dispatch(clearApiError()) as unknown as void;
  }, [dispatch]);
```

Don't love it but the type of the return sig of useEffect doesn't seem to be able to infer the return correctly there.

Reply:
This works but I do have to cast the return value since the hook can't infer it here
`useEffect(() => () => dispatch(clearApiError()) as unknown as void, [dispatch]);`